ledger/participant-integration-api/src/test/suite/scala/platform/indexer/RecoveringIndexerSpec.scala

```diff
@@ -147,41 +151,96 @@ final class RecoveringIndexerSpec
-            EventSubscribeFail("A"),
-            EventSubscribeCalled("B"),
-            EventSubscribeSuccess("B"),
-            EventStreamFail("B"),
-            EventStopCalled("B"),
-            EventSubscribeCalled("C"),
-            EventSubscribeSuccess("C"),
-            EventStreamComplete("C"),
-            EventStopCalled("C"),
-          )
-          readLog() should contain theSameElementsInOrderAs Seq(
-            Level.INFO -> "Starting Indexer Server",
-            Level.ERROR -> "Error while starting indexer, restart scheduled after 10 milliseconds",
-            Level.INFO -> "Restarting Indexer Server",
-            Level.INFO -> "Restarted Indexer Server",
-            Level.ERROR -> "Error while running indexer, restart scheduled after 10 milliseconds",
-            Level.INFO -> "Restarting Indexer Server",
-            Level.INFO -> "Restarted Indexer Server",
-            Level.INFO -> "Successfully finished processing state updates",
-            Level.INFO -> "Stopping Indexer Server",
-            Level.INFO -> "Stopped Indexer Server",
-          )
-          testIndexer.openSubscriptions shouldBe mutable.Set.empty
-        }
+      for {
+        (healthReporter, complete) <- resource.asFuture
+        _ <- complete.transformWith(finallyRelease(resource))
+      } yield {
+        testIndexer.actions shouldBe Seq[IndexerEvent](
+          EventSubscribeCalled("A"),
+          EventSubscribeFail("A"),
+          EventSubscribeCalled("B"),
+          EventSubscribeSuccess("B"),
+          EventStreamFail("B"),
+          EventStopCalled("B"),
+          EventSubscribeCalled("C"),
+          EventSubscribeSuccess("C"),
+          EventStreamComplete("C"),
+          EventStopCalled("C"),
+        )
+        readLog() should contain theSameElementsInOrderAs Seq(
+          Level.INFO -> "Starting Indexer Server",
+          Level.ERROR -> "Error while starting indexer, restart scheduled after 10 milliseconds",
+          Level.INFO -> "Restarting Indexer Server",
+          Level.INFO -> "Restarted Indexer Server",
+          Level.ERROR -> "Error while running indexer, restart scheduled after 10 milliseconds",
+          Level.INFO -> "Restarting Indexer Server",
+          Level.INFO -> "Restarted Indexer Server",
+          Level.INFO -> "Successfully finished processing state updates",
+          Level.INFO -> "Stopping Indexer Server",
+          Level.INFO -> "Stopped Indexer Server",
+        )
+        testIndexer.openSubscriptions shouldBe mutable.Set.empty
+        healthReporter.currentHealth() shouldBe HealthStatus.healthy
+      }
+    }
+
+    "report correct health status updates on failures and recoveries" in {
+      val healthStatusLogCapture = mutable.ArrayBuffer.empty[HealthStatus]
+      val healthStatusRef = new AtomicReference[HealthStatus]()
+
+      val recoveringIndexer =
+        new RecoveringIndexer(
+          actorSystem.scheduler,
+          actorSystem.dispatcher,
+          10.millis,
```

Comment:
You might want to specify this as a constant e.g. `someTimeout` so you don't have to repeat it 10+ times in this file.

Reply:
Done